x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/regex/LikePattern.java

```diff
@@ -48,6 +48,16 @@ public String asJavaRegex() {
         return regex;
     }
 
+    @Override
+    public boolean matchesAll() {
+        for (int i = 0 ; i < pattern.length(); i++) {
+            if (pattern.charAt(i) != '%') {
```

Comment:
I think that now the "problem" is reversed since the ReplaceWildcards takes place before this rule, but the test seems to work: https://github.com/elastic/elasticsearch/pull/62682/files#diff-6dd3fd1d0809ee5978cdccbc6bb7d7e1R17 ? In any case I think we should have a comment mentioning the order of the rules are applied and the choice of % or * in this rule.

Reply:
`LikePattern` has several styles. In its constructor it accepts the `%` and `_` characters instead of `*` and `?`. The conversion is done internally and accessible.
The check above works on the raw pattern - I'll follow-up with a comment.